src/Features/VisualBasic/Portable/Structure/Providers/MultiLineIfBlockStructureProvider.vb

```diff
@@ -14,9 +14,59 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
                                                   spans As ArrayBuilder(Of BlockSpan),
                                                   options As OptionSet,
                                                   cancellationToken As CancellationToken)
-            spans.AddIfNotNull(CreateBlockSpanFromBlock(
-                               node, node.IfStatement, autoCollapse:=False,
-                               type:=BlockTypes.Conditional, isCollapsible:=True))
+
+            ' The full block: eg If ... Then ... End If
+            spans.AddIfNotNull(CreateBlockSpanFromBlock(node, node.IfStatement,
+                                                        autoCollapse:=False,
+                                                        type:=BlockTypes.Conditional,
+                                                        isCollapsible:=True))
+
+            ' Attempt to make the code block after the If...Then collapsable.
+            Make_IfThenCodeBlockCollapsible(node, spans)
+
+            ' Make the ElseIfs collapsable.
+            Make_ElseIfCodeBlocksCollapsible(node, spans)
+
+            ' Make the Else section collapsable.
+            Make_ElseCodeBlockCollapsible(node, spans)
+
+        End Sub
+
+        Private Shared Sub Make_ElseCodeBlockCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            If node.ElseBlock IsNot Nothing Then
+                spans.AddIfNotNull(CreateBlockSpanFromBlock(node.ElseBlock, "Else ... ",
+                                                            autoCollapse:=False,
+                                                            type:=BlockTypes.Conditional,
+                                                            isCollapsible:=True))
+            End If
+        End Sub
+
+        Private Shared Sub Make_ElseIfCodeBlocksCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            For Each [ElseIf] In node.ElseIfBlocks
+                spans.AddIfNotNull(CreateBlockSpanFromBlock([ElseIf], [ElseIf].ElseIfStatement,
+                                                            autoCollapse:=False,
+                                                            type:=BlockTypes.Conditional,
+                                                            isCollapsible:=True))
+            Next
+        End Sub
+
+        Private Shared Sub Make_IfThenCodeBlockCollapsible(node As MultiLineIfBlockSyntax, spans As ArrayBuilder(Of BlockSpan))
+            Dim statements = node.Statements
+            If statements.Count > 0 Then
+                Dim starting = node.Statements(0).GetLeadingTrivia.Where(Function(t) t.IsWhitespaceOrEndOfLine Or t.IsRegularOrDocComment)
```

Comment:
why do we need this?  do we do this elsewhere with other VB outlining constructs?